tests/test_lora.py

```diff
@@ -524,6 +528,9 @@ def test_against_hf_mixtral():
         n_query_groups=2,
         intermediate_size=86,
         n_expert=4,
+        lora_r=1,
+        lora_key=True,
+        lora_value=True,
```

Comment:
What's the motivation for adding these? Shouldn't the end to end results fail now?

Reply:
By default, `lora_r` is 0 which basically makes it a regular GPT, not LoRA variant, since we skip LoRA path.

Reply:
Right but the test compares the numerical output to the regular model from HuggingFace

Reply:
Yes :)
The `lora_B` weights are initialized with zeros, so `lora_A` @ `lora_B` produces a matrix filled with zeros.
Pretrained + [0, 0, ..., 0] = Pretrained, thus it allows us to compare to the regular model.
Here we check this assumption plus verify that the code works properly.

Reply:
Ahh I forgot that the LoRA __init__ calls `reset_parameters`. Thank you LoRA expert